APSHTTPClient/APSHTTPRequest.m

```diff
@@ -321,9 +322,50 @@ -(void)connection:(NSURLConnection *)connection willSendRequestForAuthentication
 -(void)URLSession:(NSURLSession *)session task:(NSURLSessionTask *)task didReceiveChallenge:(NSURLAuthenticationChallenge *)challenge completionHandler:(void (^)(NSURLSessionAuthChallengeDisposition, NSURLCredential * _Nullable))completionHandler
 {
     DebugLog(@"%s", __PRETTY_FUNCTION__);
+    BOOL useSubDelegate = (self.connectionDelegate != nil && [self.connectionDelegate respondsToSelector:@selector(URLSession:task:didReceiveChallenge:completionHandler:)]);
+    
+    if(useSubDelegate && [self.connectionDelegate respondsToSelector:@selector(willHandleChallenge:forSession:)]) {
+        useSubDelegate = [self.connectionDelegate willHandleChallenge:challenge forSession:session];
+    }
+    
+    if(useSubDelegate) {
+        @try {
+            [self.connectionDelegate URLSession:session task:task didReceiveChallenge:challenge completionHandler:completionHandler];
+        }
+        @catch (NSException *exception) {
+            if (self.task != nil) {
+                [self.task cancel];
+                
+                NSMutableDictionary *dictionary = nil;
+                if (exception.userInfo) {
+                    dictionary = [NSMutableDictionary dictionaryWithDictionary:exception.userInfo];
+                } else {
+                    dictionary = [NSMutableDictionary dictionary];
```

Comment:
The dictionary should remain `nil` if there is no user-info.

Reply:
This is copied from previous NSURLConnection.